test/e2e/ConstantProduct.t.sol

```diff
@@ -118,4 +118,90 @@ contract E2EConditionalOrderTest is BaseComposableCoWTest {
         assertEq(DAI.balanceOf(owner), endBalanceDai);
         assertEq(WETH.balanceOf(owner), endBalanceWeth);
     }
+
+    function testE2ECustomOrder() public {
+        uint256 startAmountDai = 2_000 ether;
+        uint256 startAmountWeth = 1 ether;
+        // Deal the AMM reserves to the owner.
+        deal(address(DAI), address(owner), startAmountDai);
+        deal(address(WETH), address(owner), startAmountWeth);
+
+        vm.startPrank(owner);
+        DAI.approve(address(ammFactory), type(uint256).max);
+        WETH.approve(address(ammFactory), type(uint256).max);
+
+        // Funds have been allocated.
+        assertEq(DAI.balanceOf(owner), startAmountDai);
+        assertEq(WETH.balanceOf(owner), startAmountWeth);
+
+        uint256 minTradedToken0 = 0;
+        bytes memory priceOracleData = abi.encode(UniswapV2PriceOracle.Data(pair));
+        bytes32 appData = keccak256("order app data");
+        ConstantProduct amm = ammFactory.create(
+            DAI, startAmountDai, WETH, startAmountWeth, minTradedToken0, uniswapV2PriceOracle, priceOracleData, appData
+        );
+        vm.stopPrank();
+
+        // Funds have been transferred to the AMM.
+        assertEq(DAI.balanceOf(owner), 0);
+        assertEq(WETH.balanceOf(owner), 0);
+
+        ConstantProduct.TradingParams memory data = ConstantProduct.TradingParams({
+            minTradedToken0: minTradedToken0,
+            priceOracle: uniswapV2PriceOracle,
+            priceOracleData: priceOracleData,
+            appData: appData
+        });
+        uint256 sellAmount = 100 ether;
+        uint256 buyAmount = 1 ether;
+
+        // The trade will be settled against bob.
+        deal(address(DAI), bob.addr, startAmountDai);
+        deal(address(WETH), bob.addr, startAmountWeth);
+        vm.startPrank(bob.addr);
+        DAI.approve(address(relayer), type(uint256).max);
+        WETH.approve(address(relayer), type(uint256).max);
+        vm.stopPrank();
+
+        GPv2Order.Data memory order = GPv2Order.Data({
+            sellToken: DAI,
+            buyToken: WETH,
+            receiver: GPv2Order.RECEIVER_SAME_AS_OWNER,
+            sellAmount: sellAmount,
+            buyAmount: buyAmount,
+            validTo: uint32(block.timestamp) + 1,
```

Comment:
If the primary purpose of the E2E test is to show solvers how to solve using a JIT order against a CoW AMM, they should also perhaps have some guidance around the application of `validTo` - primarily that there is a `MAX_ORDER_DURATION` and not setting this value to something sane (ie. just bumping it from now timestamp + 1s) increases reversion risk / guarantees reversion.